Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -966,8 +966,23 @@ void NetworkConnectionToWebProcess::registerInternalFileBlobURL(const URL& url,
     if (!session)
         return;
 
-    m_blobURLs.add({ url, std::nullopt });
-    session->blobRegistry().registerInternalFileBlobURL(url, BlobDataFileReferenceWithSandboxExtension::create(path, replacementPath, SandboxExtension::create(WTFMove(extensionHandle))), contentType);
+    auto after = [this, weakThis = WeakPtr { * this }, url, path, replacementPath, extensionHandle = WTFMove(extensionHandle), contentType] (bool allowed) mutable {
```

Comment:
Can we call this completionHandler or registerInternalFileBlobURLs or continueRegistration or something?